parquet-column/src/main/java/org/apache/parquet/column/impl/ColumnReaderImpl.java

```diff
@@ -546,6 +573,11 @@ private void initDataReader(Encoding dataEncoding, byte[] bytes, int offset, int
     } catch (IOException e) {
       throw new ParquetDecodingException("could not read page in col " + path, e);
     }
+
+    if (dataEncoding.requiresSequentialReads(writerVersion) && previousReader != null) {
```

Comment:
This is more generic. This allows the data encoding to require sequential reads for any reason, which will result in the previous values reader being passed to the new values reader. This is a generic way to make that happen in case we run into it in the future.